users/selfmanagedcredswithoutmfa_policy.tf

```diff
@@ -0,0 +1,149 @@
+# ------------------------------------------------------------------------------
+# IAM policy that allows users to administer their own user
+# accounts.  This policy is pretty much copied from here:
+# https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html
+#
+# The main difference is that we don't require these users to be
+# authenticated with MFA, since these accounts will be accessed
+# programatically (i.e. not via the AWS web console) where MFA is not
+# an option.
+#
+# Note that IAM policy variables (e.g. "aws:username") are used below. For
+# reference, refer to:
+# https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html
+# https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "self_managed_creds_without_mfa" {
+  # Allow users to view their own account information
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:GetAccountPasswordPolicy",
+      "iam:GetAccountSummary",
+      "iam:ListVirtualMFADevices",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  # Allow users to administer their own passwords
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:ChangePassword",
+      "iam:GetUser",
+    ]
+
+    resources = [
+      "arn:aws:iam::*:user/&{aws:username}",
+    ]
+  }
+
+  # Allow users to administer their own access keys
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:CreateAccessKey",
+      "iam:DeleteAccessKey",
+      "iam:ListAccessKeys",
+      "iam:UpdateAccessKey",
+    ]
+
+    resources = [
+      "arn:aws:iam::*:user/&{aws:username}",
+    ]
+  }
+
+  # Allow users to administer their own signing certificates
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:DeleteSigningCertificate",
+      "iam:ListSigningCertificates",
+      "iam:UpdateSigningCertificate",
+      "iam:UploadSigningCertificate",
+    ]
+
+    resources = [
+      "arn:aws:iam::*:user/&{aws:username}",
+    ]
+  }
+
+  # Allow users to administer their own ssh public keys
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "iam:DeleteSSHPublicKey",
+      "iam:GetSSHPublicKey",
+      "iam:ListSSHPublicKeys",
+      "iam:UpdateSSHPublicKey",
+      "iam:UploadSSHPublicKey",
+    ]
+
+    resources = [
+      "arn:aws:iam::*:user/&{aws:username}",
+    ]
+  }
+
+  # Allow users to administer their own git credentials
```

Comment:
Is this truly just for `git`.

Reply:
The only use I know of is AWS CodeCommit.

Reply:
From [this page](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html):

> [this] statement allows the user to create, update, and delete their own Git credentials for CodeCommit.

We don't yet use this feature anywhere either, as far as I am aware.

More info here:
* https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateServiceSpecificCredential.html
* https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_ssh-keys.html